xc/device.ml

```diff
@@ -1444,22 +1457,34 @@ module Vusb = struct
       end
     in
     let is_running = Qemu.is_running ~xs domid in
-    match is_running with
-    | true  ->
-      let cmd = Qmp.(Device_add
-                       ( "usb-host"
-                       , id
-                       , Some (get_bus version, hostbus, hostport)
-                       ))
-      in qmp_send_cmd domid cmd |> ignore
-    | _ -> ()
-
-  let vusb_unplug ~xs ~domid ~id =
+    if is_running then
+      begin
+        (* Need to reset USB device before passthrough to vm according to CP-24616.
+           Also need to do deprivileged work in usb_reset script if QEMU is deprivileged.
+        *)
+        begin match Qemu.pid ~xs domid with
+          | Some pid -> call_usb_reset hostbus hostport domid pid "attach" priv
+          | _ -> failwith "qemu pid does not exist"
```

Comment:
Poor error message - provide more context - at least the domid.

Reply:
updated